From 02b7fc388fb813be03668eeaca27b65a6c4fcd8b Mon Sep 17 00:00:00 2001
From: Gatien Chevallier <gatien.chevallier@foss.st.com>
Date: Fri, 3 May 2024 17:07:39 +0200
Subject: [PATCH 0821/1695] drivers: stm32_tamp: force the use of
 st,backup-zones DT property

Force the presence of the "st,backup-zones" property in the device
tree as the backup registers configuration is mandatory.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 6f90367eee1261c4528000d33b4a0ac415cf92d7 drivers: stm32_tamp: force the use of st,backup-zones DT property]
Change-Id: I71ec5147a612eb63da232b26148fca727ae6a41d
Signed-off-by: Gatien Chevallier <gatien.chevallier@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/378281
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Reviewed-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
Tested-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/378787
---
 core/drivers/stm32_tamp.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/core/drivers/stm32_tamp.c b/core/drivers/stm32_tamp.c
index 3577972a5..1ff83ac55 100644
--- a/core/drivers/stm32_tamp.c
+++ b/core/drivers/stm32_tamp.c
@@ -924,7 +924,7 @@ static void parse_bkpregs_dt_conf(struct stm32_tamp_platdata *pdata,
 
 	cuint = fdt_getprop(fdt, node, "st,backup-zones", &lenp);
 	if (!cuint) {
-		DMSG("Default configuration for backup registers");
+		panic("Missing backup registers configuration");
 		return;
 	}
 
-- 
2.47.2

